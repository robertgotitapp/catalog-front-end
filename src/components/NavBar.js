import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import { signOut } from '../actions/users'

class NavBar extends Component {

    handleSignOut = (e) => {
        e.preventDefault()
        this.props.signOut()
    }

    render() {
        return (
            <div>
                <button> <Link to='/signin'> Sign In </Link> </button>
                <button> <Link to='/signup'> Sign Up </Link> </button>
                <button onClick={this.handleSignOut}>Sign Out</button>
            </div>
        )
    }
}

const mapDispatchtoProps = {
    signOut
}

export default connect(null, mapDispatchtoProps)(NavBar)