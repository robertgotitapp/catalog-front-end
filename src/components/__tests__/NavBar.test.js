import React from 'react';
import { shallow } from 'enzyme';
import { NavBar } from '../NavBar';

describe('components/NavBar', () => {
  describe('when user is logged in', () => {
    const props = {
      isLoggedIn: true,
    };
    const container = shallow(<NavBar {...props} />);

    it('should render correctly', () => {
      expect(container).toMatchSnapshot();
    });

    it('should have 1 Add Category Link', () => {
      expect(container.find('[name="addcategory"]')).toHaveLength(1);
    });

    it('should have 1 Home Link', () => {
      expect(container.find('[name="home"]')).toHaveLength(1);
    });

    it('should have 1 Add Item Link', () => {
      expect(container.find('[name="additem"]')).toHaveLength(1);
    });

    it('should trigger handler when Sign Out is clicked', () => {
      container.instance().handleSignOut = jest.fn();
      container.instance().forceUpdate();
      const signOutLink = container.find('[name="signout"]');
      signOutLink.simulate('click');
      expect(container.instance().handleSignOut).toHaveBeenCalled();
    });
  });

  describe('when user is not logged in', () => {
    const props = {
      isLoggedIn: false,
    };
    const container = shallow(<NavBar {...props} />);

    it('should render correctly', () => {
      expect(container).toMatchSnapshot();
    });

    it('should have 1 Sign In Link', () => {
      expect(container.find('[name="signin"]')).toHaveLength(1);
    });

    it('should have 1 Home Link', () => {
      expect(container.find('[name="home"]')).toHaveLength(1);
    });

    it('should have 1 Sign Out Link', () => {
      expect(container.find('[name="signup"]')).toHaveLength(1);
    });
  });
});
