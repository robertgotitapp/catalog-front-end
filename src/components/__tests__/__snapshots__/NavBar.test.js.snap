// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/NavBar when user is logged in should render correctly 1`] = `
<div>
  <Bootstrap(undefined)
    bg="light"
    variant="light"
  >
    <NavbarBrand>
      <img
        alt="Got It Logo"
        className="d-inline-block align-top logo"
        src="logo.jpeg"
      />
    </NavbarBrand>
    <Nav
      className="mr-auto"
      fill={false}
      justify={false}
    >
      <NavItem
        className="navItem"
      >
        <Link
          name="home"
          to="/"
        >
          Home
        </Link>
      </NavItem>
      <NavItem
        className="navItem"
      >
        <Link
          name="addcategory"
          to="/newcategory"
        >
          Add Category
        </Link>
      </NavItem>
      <NavItem
        className="navItem"
      >
        <Link
          name="additem"
          to="/newitem"
        >
          Add Item
        </Link>
      </NavItem>
      <NavItem
        className="navItem"
      >
        <Link
          name="signout"
          onClick={[Function]}
          to="/"
        >
          Sign Out
        </Link>
      </NavItem>
    </Nav>
  </Bootstrap(undefined)>
</div>
`;

exports[`components/NavBar when user is not logged in should render correctly 1`] = `
<div>
  <Bootstrap(undefined)
    bg="light"
    variant="light"
  >
    <NavbarBrand>
      <img
        alt="Got It Logo"
        className="d-inline-block align-top logo"
        src="logo.jpeg"
      />
    </NavbarBrand>
    <Nav
      className="mr-auto"
      fill={false}
      justify={false}
    >
      <NavItem
        className="navItem"
      >
        <Link
          name="home"
          to="/"
        >
          Home
        </Link>
      </NavItem>
      <NavItem
        className="navItem"
      >
        <Link
          name="signin"
          to="/signin"
        >
          Sign In
        </Link>
      </NavItem>
      <NavItem
        className="navItem"
      >
        <Link
          name="signup"
          to="/signup"
        >
          Sign Up
        </Link>
      </NavItem>
    </Nav>
  </Bootstrap(undefined)>
</div>
`;
